src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -166,6 +167,11 @@ namespace AppInstaller::CLI
                 ReportSearchResultIdentity <<
                 GetInstalledPackageVersion;
 
+            if (context.IsTerminated())
+            {
+                return;
+            }

```

Comment:
This should never be needed, as the rest of the output operators will simply be skipped.

Reply:
The below:
 context << SelectLatestApplicableUpdate(*(context.Get<Execution::Data::SearchResult>().Matches.at(0).Package));

will construct the workflowtask first which throws invalid vector position 

Do you want to check size of context.Get<Execution::Data::SearchResult>().Matches instead below?

---
In reply to: [518308965](https://github.com/microsoft/winget-cli/pull/634#discussion_r518308965) [](ancestors = 518308965)